[10.0] [MIG] Purchase Request Procurement (purchase_request_procurement)

ThomasBinsfeld opened the pull request:
Migration of module purchase_request_procurement to 10.0
Depends on #424

ThomasBinsfeld committed:
[MIG] purchase_request_procurement to 10.0

ThomasBinsfeld committed:
[IMP] Rename files

ThomasBinsfeld committed:
[ADD] setup.py

LoisRForgeFlow commented on purchase_request_procurement/models/procurement_order.py:
```diff
@@ -0,0 +1,95 @@
+    )
+
+    @api.model
+    def _prepare_purchase_request_line(self, purchase_request, procurement):
```
why not to remove procurement from all the inputs and use self instead?

LoisRForgeFlow commented on purchase_request_procurement/models/procurement_order.py:
```diff
@@ -0,0 +1,95 @@
+        }
+
+    @api.model
+    def _prepare_purchase_request(self, procurement):
```
comment above

LoisRForgeFlow commented on purchase_request_procurement/models/procurement_order.py:
```diff
@@ -0,0 +1,95 @@
+        }
+
+    @api.model
+    def _search_existing_purchase_request(self, procurement):
```
same

LoisRForgeFlow commented on purchase_request_procurement/test/purchase_request_from_procurement.yml:
```diff
@@ -21,11 +21,11 @@
-        assert line.product_id.id == procurement.product_id.id, "Product is not correspond."
-        assert line.product_uom_id.id == procurement.product_uom.id, "UOM is not correspond."
-        assert line.product_qty == procurement.product_qty, "Quantity is not correspond."
+        assert line.product_id.id == procurement.product_id.id, "Product does not correspond."
```
I think there is not goal in mantaining the yml tests in v10. This is a good moment to get rid of this IMO.

LoisRForgeFlow reviewed (commented):
Tested on runbot and it is fine, just some comments in lines.

ThomasBinsfeld committed:
[MIG] purchase_request_procurement: remove .yml tests

ThomasBinsfeld committed:
[IMP] purchase_request_procurement: make api.model methods api.multi
The two prepare methods and the serac existing purchase request method can be api.multi

ThomasBinsfeld commented:
@lreficent Thanks for the review. Done !

LoisRForgeFlow reviewed (approved):
Code review of new changes. LGTM now :+1:

benwillig committed:
[REF] Refactored create purchase request process to be able to inherit easily

LoisRForgeFlow committed:
[9.0][IMP] purchase_request_procurement: adapt to cancelled state of lines

benwillig committed:
[[10.0][IMP] Cancel procurements when rejecting a purchase request

MiquelRForgeFlow reviewed (approved):
LGTM :+1:

AaronHForgeFlow commented:
Code LGTM. Pending functional tests.

LoisRForgeFlow commented:
@ThomasBinsfeld Could you check the changes proposed to v9? (see: https://github.com/OCA/purchase-workflow/pull/398 and https://github.com/OCA/purchase-workflow/pull/412)

benwillig commented:
This PR now depends on #424.

ThomasBinsfeld commented:
Ping @lreficent

LoisRForgeFlow commented:
LGTM :+1: 

Just a minor comment, the commit is kind of a cherry-pick of the changes I proposed in v9, is it possible to keep the authority of the commit?

Thanks for the work!

benwillig committed:
[REF] Removed deprecated data xml node

benwillig commented:
Ups, I forgot to continue my cherry pick :laughing:. It's fixed now.

LoisRForgeFlow commented:
Thanks! :smile: 

Once the issues with travis are fixed I think this will be ready.

GillesTephaneMeyomesse committed:
[FIX] flake8

AaronHForgeFlow reviewed (changes requested):
It fails when a procurement is cancelled :
![image](https://user-images.githubusercontent.com/19620251/27905040-006175a8-623f-11e7-8a38-2e584be201e2.png)

benwillig commented:
@aheficent At the moment, PR #424 has not been merged yet. That's why it doesn't work.

AaronHForgeFlow commented:
@benwillig I saw. Let's get that one merged first then. Thank you for telling me.

LoisRForgeFlow commented:
@ThomasBinsfeld Would you fix conflicts and add this v9 fix made here: https://github.com/OCA/purchase-workflow/commit/08c510912c89489be2c2235e1e96cc6d5826b17e?

edit: the fix was done here: https://github.com/OCA/purchase-workflow/pull/441

LoisRForgeFlow committed:
[9.0][FIX] purchase_request_procurement: interfering mrp_production_request
From https://github.com/OCA/purchase-workflow/commit/08c510912c89489be2c2235e1e96cc6d5826b17e

ThomasBinsfeld commented:
Rebased on 10.0

LoisRForgeFlow commented:
Great, thanks! Just the same that in previous cherry-pick: try to keep the authorship.

ThomasBinsfeld commented:
@lreficent Authorship restored ! :white_check_mark:

D4N commented:
#424 has been merged, so could this PR be merged, too? @aheficent Had some comments, but I guess they have been fixed, as #424 is merged.
If there are still issues, I offer my help, as I would really like this module in v10.

AaronHForgeFlow reviewed (approved):
Tested on runbot :+1:

JordiBForgeFlow commented:
Thanks to all for your efforts!!
